### DISCUSSION TITLE
Add a status indicator when a user is in the song select menu during multiplayer lobbies

### DISCUSSION THREAD

**Discussion by marioman63** [ppy/osu#24293] [Ideas]
When in a Laser multiplayer lobby, there is an indicator in the form of text and an icon showing when players are missing a map, spectating, playing, downloading, ready to start, etc. However, there is currently no way to tell if a player is viewing the song select screen.

In it's current state, it is impossible to tell if a user is AFK or browsing the song select, leading to some awkward interactions between users due to this confusion. In my experience this results in unruly behaviour in chat trying to get said user's attention, which doesn't work of course due to the lack of chat in the song select screen (but that is a separate concern). It can also result in unintentional starts because someone was picking music for the queue, unbeknownst to the host.

While I am unsure of the amount of work it would take to add such a status indicator, I think it would be helpful in communicating to the room what a user is up to which would alleviate the above concerns.

**Comment by peppy** [comment]
Definitely something we want. Shouldn't be a huge amount of work to implement.

**Comment by GerritPlehn** [comment]
Do you think this is a good task for a first time contributor?

My general approach would be to add a new boolean property to the `MultiplayerRoomUser`, since a user could be in `MultiplayerUserState`'s `Ready` or `Spectating` and choosing a map at the same time.
Then toggling that bool from `MultiplayerMatchSubScreen.OpenSongSelection` in a similar matter to the other states through the `MultiplayerClient`.
Does that sound reasonable to you?

**Reply by peppy** [reply]
I'm not sure it makes sense for a user to be both "ready" and "selecting beatmap". Maybe the ready flag should be removed.

**Reply by GerritPlehn** [reply]
I see your point. For ready it really doesn't make much sense. Then I'd keep the "Spectating" state intact, so that players won't be pulled into the match as a player when it starts while they're picking a map. And replace "Ready" with "Choosing Map" if it's set.
I will give that approach a shot. Thanks for the feedback.

**Reply by peppy** [reply]
See how you go. If the logic gets convoluted implementing that way, I'd be okay with a second enum that's like `CurrentScreen`. This would allow for potential future cases like "showing the user is changing room settings" or the likes. Definitely not a `bool` though.
